modules/artifact/src/main/java/org/mule/runtime/module/artifact/api/descriptor/DeployableArtifactDescriptor.java

```diff
@@ -40,6 +42,7 @@ public class DeployableArtifactDescriptor extends ArtifactDescriptor {
   private File logConfigFile;
   private Optional<Properties> deploymentProperties = empty();
   private Set<String> supportedJavaVersions = emptySet();
+  private final String identifier = valueOf(currentTimeMillis());
```

Comment:
Lets use an UUID to avoid collisions if (for any reason) the folders do no get deleted.